stackprinter/formatting.py

```diff
@@ -163,15 +154,15 @@ def format_exc_info(etype, evalue, tb, style='plaintext', add_summary='auto',
         # Now, actually do some formatting:
         msgs = []
         if tb:
-            frameinfos = [ex.get_info(tb_) for tb_ in _walk_traceback(tb)]
```

Comment:
the reason for calling get_info out here outside the formatter is that I wanted to render the main traceback and the summary traceback from the same data, without re-analyzing the source (in fact that's the entire reason why the formatters accept predigested frameinfos)
it's not _entirely_ just optimization for optimization's sake, the `show`-call in `demo_complex.py` takes ~75ms on master, ~100ms on this branch